x-pack/plugins/observability/public/hooks/slo/use_fetch_slo_list.ts

```diff
@@ -84,7 +84,7 @@ export function useFetchSloList({
           return;
         }
 
-        if (results.find((slo) => slo.summary.status === 'NO_DATA')) {
+        if (results.find((slo) => slo.summary.status === 'NO_DATA' || !slo.summary)) {
```

Comment:
🍰 nit: extract this condition into a well-named function/variable (can be define just above) to make more explicit what this condition is checking, e.g. `containsSloWithoutData`, `someSloWithoutData` or something along this lines.